src/ProjectOrigin.WalletSystem.IntegrationTests/CommandHandlers/VerifySliceCommandHandlerTests.cs

```diff
@@ -0,0 +1,365 @@
+using ProjectOrigin.PedersenCommitment;
+using ProjectOrigin.WalletSystem.Server.Activities.Exceptions;
+using ProjectOrigin.WalletSystem.Server.CommandHandlers;
+using ProjectOrigin.WalletSystem.Server.Database;
+using ProjectOrigin.WalletSystem.Server.Models;
+using ProjectOrigin.WalletSystem.Server.Projections;
+using ProjectOrigin.WalletSystem.Server.Repositories;
+using ProjectOrigin.WalletSystem.Server.Services;
+using System;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace ProjectOrigin.WalletSystem.IntegrationTests.CommandHandlers;
+
+public class VerifySliceCommandHandlerTests : IAsyncLifetime
+{
+    const string RegistryName = "MyRegistry";
+    const string Area = "Narnia";
+
+    private Fixture _fixture;
+    private ICertificateRepository _certificateRepository;
+    private IWalletRepository _walletRepository;
+    private ILogger<VerifySliceCommandHandler> _logger;
+    private IRegistryService _registryService;
+    private ServiceProvider _provider;
+
+    public VerifySliceCommandHandlerTests()
+    {
+        _fixture = new Fixture();
+
+        _certificateRepository = Substitute.For<ICertificateRepository>();
+        _walletRepository = Substitute.For<IWalletRepository>();
+        _logger = Substitute.For<ILogger<VerifySliceCommandHandler>>();
+
+        var unitOfWork = Substitute.For<IUnitOfWork>();
+        unitOfWork.WalletRepository.Returns(_walletRepository);
+        unitOfWork.CertificateRepository.Returns(_certificateRepository);
+
+        _registryService = Substitute.For<IRegistryService>();
+        _provider = new ServiceCollection()
+            .AddSingleton(unitOfWork)
+            .AddSingleton(_registryService)
+            .AddSingleton(_logger)
+            .AddMassTransitTestHarness(x =>
+            {
+                x.AddConsumer<VerifySliceCommandHandler>();
+            })
+            .BuildServiceProvider(true);
+    }
+
+    public Task InitializeAsync()
+    {
+        return Task.CompletedTask;
+    }
+
+    public async Task DisposeAsync()
+    {
+        await _provider.DisposeAsync();
+    }
+
+    [Fact]
+    public async void WhenReceivedSliceIsValid_ExpectIsConvertedToSliceAndCertificateIsCreated()
+    {
+        // Arrange
+        var certId = Guid.NewGuid();
+        IHDPrivateKey privateKey = new Secp256k1Algorithm().GenerateNewPrivateKey();
+        var walletPosition = 1;
+        var depositPosition = 1;
+        var commitment = new SecretCommitmentInfo(150);
+
+        var depositEndpoint = new DepositEndpoint(Guid.NewGuid(), Guid.NewGuid(), walletPosition, privateKey.Derive(walletPosition).Neuter(), string.Empty, string.Empty, string.Empty);
+        _walletRepository.GetDepositEndpoint(depositEndpoint.Id).Returns(depositEndpoint);
+
+        IssuedEvent issuedEvent = CreateIssuedEvent(certId, commitment, privateKey.Derive(walletPosition).Derive(depositPosition).PublicKey);
```

Comment:
It may be hard to understand `privateKey.Derive(walletPosition).Derive(depositPosition).PublicKey`

Reply:
fix